src/worker/worker.c

```diff
@@ -263,6 +288,14 @@ static void kill_job(struct nm_event_execution_properties *event)
 		(void)kill(-cp->ei->pid, SIGKILL);
 		return;
 	}
+	/* check if the child we'r killing belongs to this worker process */
+	wpid = getpid();
+	get_process_parent_id(pid, &ppid);
+	if (ppid == -1 || ppid != wpid) {
```

Comment:
Now if the function fails this evaluates to `true`. Ah, by the way: if `ppid == -1`  it will always be `!= wpid`. Am I wrong?

Reply:
Yes, if we fail to find the pid we were looking for this will evaluate to `true` so we can return early. I think you're right with your last statement so perhaps the first evaluation is unnecessary. Although I'm fine with the way it is now. Makes it somewhat easier to read.